### ISSUE TITLE
Switching to Windows container pops an error dialog

### ISSUE THREAD

**Issue by HuanhuanSunMSFT** [docker/for-win#178]
### Expected behavior

You can switch to Windows container
### Actual behavior

An error dialog pops up:

```
Error response from daemon: i/o timeout

   at Docker.Backend.DockerDaemonChecker.Check()
   at Docker.Backend.ContainerEngine.Windows.DoStart(Settings settings)
   at Docker.Backend.ContainerEngine.Windows.Start(Settings settings)
   at Docker.Core.Pipe.NamedPipeServer.<>c__DisplayClass8_0.<Register>b__0(Object[] parameters)
   at Docker.Core.Pipe.NamedPipeServer.RunAction(String action, Object[] parameters)
```

However, you can still run docker commands, such as:
docker version:

```
Client:
 Version:      1.12.3
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   6b644ec
 Built:        Thu Oct 27 00:09:21 2016
 OS/Arch:      windows/amd64
 Experimental: true

Server:
 Version:      1.12.2-cs2-ws-beta
 API version:  1.25
 Go version:   go1.7.1
 Git commit:   050b611
 Built:        Tue Oct 11 02:35:40 2016
 OS/Arch:      windows/amd64
```

You can also create and start a windows container. It's just the icon in system tray is always red after above dialog.

Restart Docker for Windows sometimes fixes it.
### Information
- Diagnostic ID: FBB60744-F952-4741-8BDC-7DEE1DBA5E93/2016-10-28_10-59-48
- Error logs:

```
Version: 1.12.3-beta29.2 (8280)
Channel: Beta
Sha1: 902414df0cea7fdc85b87f0077b0106c3af9f64c
Started on: 2016/10/28 10:59:00.135
Resources: C:\Program Files\Docker\Docker\Resources
OS: Windows 10 Enterprise
Edition: Enterprise
Id: 1607
Build: 14393
BuildLabName: 14393.321.amd64fre.rs1_release_inmarket.161004-2338
You can send feedback, including this log file, at https://github.com/docker/for-win/issues
[10:59:00.104][Tracking       ][Info   ] Crash report and usage statistics are enabled
[10:59:00.291][SegmentApi     ][Info   ] Usage statistic: appLaunched
[10:59:00.370][NamedPipeClient][Info   ] Sending Version()...
[10:59:00.557][NamedPipeClient][Info   ] Received response for Version
[10:59:00.573][SegmentApi     ][Info   ] Usage statistic: heartbeat
[10:57:51.575][BackendServer  ][Info   ] Started
[10:57:51.622][Moby           ][Info   ] Connecting...
[10:59:00.510][NamedPipeServer][Info   ] Version()
[10:59:00.510][NamedPipeServer][Info   ] Version done.
[10:59:00.838][NamedPipeClient][Info   ] Sending Start(Docker.Core.Settings)...
[10:59:00.838][Updater        ][Info   ] Checking for updates on channel Beta...
[10:59:00.838][NamedPipeServer][Info   ] Start(Docker.Core.Settings)
[10:59:00.932][PowerShell     ][Info   ] Run script 'Import-Module Dism; Get-WindowsOptionalFeature -Online | where FeatureName -like "*Hyper-V*" | where State -ne Enabled'...
[10:59:01.401][Updater        ][Info   ] Local build 8280 is as good as the remote 8280 on channel Beta
[10:59:18.432][PowerShell     ][Info   ] Run script 'Import-Module Dism; Get-WindowsOptionalFeature -Online | where FeatureName -like "*Containers*" | where State -ne Enabled'...
[10:59:19.916][ContainerEngineHelper][Error  ] Might have failed to kill a running dockerd process: Process with an Id of 8208 is not running.
[10:59:19.932][WindowsDockerDaemon][Info   ] Starting C:\Program Files\Docker\Docker\Resources\dockerd.exe -H npipe:////./pipe/docker_engine_windows
[10:59:20.119][WindowsDockerDaemon][Info   ] Started
[10:59:20.135][ApiProxy       ][Info   ] Starting C:\Program Files\Docker\Docker\Resources\com.docker.proxy.exe -daemonNamedPipe=\\.\pipe\docker_engine_windows
[10:59:20.244][ApiProxy       ][Info   ] Started
[10:59:21.013][ApiProxy       ][Info   ] docker proxy (on deprecated port): ready
[10:59:21.019][ApiProxy       ][Info   ] docker proxy: ready
[10:59:22.327][WindowsDockerDaemon][Info   ] time="2016-10-28T10:59:22.325208100-07:00" level=info msg="Windows default isolation mode: hyperv" 
[10:59:22.333][WindowsDockerDaemon][Info   ] time="2016-10-28T10:59:22.327207600-07:00" level=info msg="[graphdriver] using prior storage driver: windowsfilter" 
[10:59:22.379][ApiProxy       ][Info   ] proxy >> GET /v1.24/containers/json
[10:59:22.383][ApiProxy       ][Info   ] Dial name pipe  \\.\pipe\docker_engine_windows
[10:59:22.430][ApiProxy       ][Info   ] i/o timeout
[10:59:22.471][ApiProxy       ][Info   ] proxy >> GET /v1.24/containers/json
[10:59:22.474][ApiProxy       ][Info   ] Dial name pipe  \\.\pipe\docker_engine_windows
[10:59:22.521][ApiProxy       ][Info   ] i/o timeout
[10:59:22.560][ApiProxy       ][Info   ] proxy >> GET /v1.24/containers/json
[10:59:22.563][ApiProxy       ][Info   ] Dial name pipe  \\.\pipe\docker_engine_windows
[10:59:22.611][ApiProxy       ][Info   ] i/o timeout
[10:59:22.644][ApiProxy       ][Info   ] proxy >> GET /v1.24/containers/json
[10:59:22.647][ApiProxy       ][Info   ] Dial name pipe  \\.\pipe\docker_engine_windows
[10:59:22.694][ApiProxy       ][Info   ] i/o timeout
[10:59:22.727][ApiProxy       ][Info   ] proxy >> GET /v1.24/containers/json
[10:59:22.730][ApiProxy       ][Info   ] Dial name pipe  \\.\pipe\docker_engine_windows
[10:59:22.777][ApiProxy       ][Info   ] i/o timeout
[10:59:22.811][ApiProxy       ][Info   ] proxy >> GET /v1.24/containers/json
[10:59:22.814][ApiProxy       ][Info   ] Dial name pipe  \\.\pipe\docker_engine_windows
[10:59:22.862][ApiProxy       ][Info   ] i/o timeout
[10:59:22.896][ApiProxy       ][Info   ] proxy >> GET /v1.24/containers/json
[10:59:22.899][ApiProxy       ][Info   ] Dial name pipe  \\.\pipe\docker_engine_windows
[10:59:22.946][ApiProxy       ][Info   ] i/o timeout
[10:59:22.998][ApiProxy       ][Info   ] proxy >> GET /v1.24/containers/json
[10:59:23.002][ApiProxy       ][Info   ] Dial name pipe  \\.\pipe\docker_engine_windows
[10:59:23.048][ApiProxy       ][Info   ] i/o timeout
[10:59:23.081][ApiProxy       ][Info   ] proxy >> GET /v1.24/containers/json
[10:59:23.084][ApiProxy       ][Info   ] Dial name pipe  \\.\pipe\docker_engine_windows
[10:59:23.132][ApiProxy       ][Info   ] i/o timeout
[10:59:23.166][ApiProxy       ][Info   ] proxy >> GET /v1.24/containers/json
[10:59:23.169][ApiProxy       ][Info   ] Dial name pipe  \\.\pipe\docker_engine_windows
[10:59:23.216][ApiProxy       ][Info   ] i/o timeout
[10:59:23.222][NamedPipeServer][Error  ] Unable to execute Start: Error response from daemon: i/o timeout
    at Docker.Backend.DockerDaemonChecker.Check()
   at Docker.Backend.ContainerEngine.Windows.DoStart(Settings settings)
   at Docker.Backend.ContainerEngine.Windows.Start(Settings settings)
   at Docker.Core.Pipe.NamedPipeServer.<>c__DisplayClass8_0.<Register>b__0(Object[] parameters)
   at Docker.Core.Pipe.NamedPipeServer.RunAction(String action, Object[] parameters)
[10:59:23.228][NamedPipeClient][Error  ] Unable to send Start: Error response from daemon: i/o timeout

[10:59:23.234][Notifications  ][Error  ] Error response from daemon: i/o timeout

[10:59:23.412][WindowsDockerDaemon][Info   ] time="2016-10-28T10:59:23.412894600-07:00" level=info msg="Graph migration to content-addressability took 0.00 seconds" 
[10:59:23.420][WindowsDockerDaemon][Info   ] time="2016-10-28T10:59:23.413896500-07:00" level=info msg="Loading containers: start." 
[10:59:23.476][NamedPipeClient][Info   ] Sending GetDebugInfo()...
[10:59:23.485][NamedPipeServer][Info   ] GetDebugInfo()
[10:59:25.727][WindowsDockerDaemon][Info   ] time="2016-10-28T10:59:25.727101100-07:00" level=error msg="Resolver Setup/Start failed for container none, \"json: cannot unmarshal array into Go value of type hcsshim.HNSNetwork\"" 
[10:59:26.269][WindowsDockerDaemon][Info   ] ......
[10:59:26.269][WindowsDockerDaemon][Info   ] time="2016-10-28T10:59:26.269889100-07:00" level=info msg="Loading containers: done." 
[10:59:26.281][WindowsDockerDaemon][Info   ] time="2016-10-28T10:59:26.270887300-07:00" level=info msg="Daemon has completed initialization" 
[10:59:26.286][WindowsDockerDaemon][Info   ] time="2016-10-28T10:59:26.270887300-07:00" level=info msg="Docker daemon" commit=050b611 graphdriver=windowsfilter version=1.12.2-cs2-ws-beta 
[10:59:26.350][WindowsDockerDaemon][Info   ] time="2016-10-28T10:59:26.350221400-07:00" level=info msg="API listen on //./pipe/docker_engine_windows" 
[10:59:47.341][NamedPipeServer][Info   ] GetDebugInfo done.
[10:59:47.355][NamedPipeClient][Info   ] Received response for GetDebugInfo
[10:59:47.394][Program        ][Info   ] Sending Bugsnag report dbd5b4cf-70c9-4f9d-9320-3379197c1147...
[10:59:48.404][Program        ][Info   ] Bugsnag report dbd5b4cf-70c9-4f9d-9320-3379197c1147 sent
[10:59:48.498][NamedPipeClient][Info   ] Sending DownloadVmLogs()...
[10:59:48.506][NamedPipeServer][Info   ] DownloadVmLogs()
[10:59:48.510][PowerShell     ][Info   ] Run script...
[10:59:48.908][HyperV         ][Warning] Unable to download logs: 2016/10/28 10:59:48 Failed to Dial e70469bf-6b44-4e53-a1c4-a643ab0e6f14 445ba2cb-e69b-4912-8b42-d7f494d007ea : The requested address is not valid in its context.

[10:59:48.916][NamedPipeServer][Info   ] DownloadVmLogs done.
[10:59:48.923][NamedPipeClient][Info   ] Received response for DownloadVmLogs
[11:08:08.695][SegmentApi     ][Info   ] Usage statistic: actionMenuAbout
[11:09:33.648][ErrorReportWindow][Info   ] Open logs
```
### Steps to reproduce the behavior
1. Install Docker for Windows version 1.12.3-beta29.2 (8280)
2. Switch to Windows Container

**Comment by alvesman** [comment]
Maybe it has something to do with this:
"But there is a little bit of confusion which sort of containers can be started with each of the two installations. And you can't run both Docker Engines side-by-side without some adjustments.
This is because each of the installations use the same default named pipe //./pipe/docker_engine causing one of the engines to fail to start."
[https://stefanscherer.github.io/run-linux-and-windows-containers-on-windows-10/](https://stefanscherer.github.io/run-linux-and-windows-containers-on-windows-10/l)

**Comment by dgageot** [comment]
@HuanhuanSunMSFT Can you please give a try to Docker for Windows Beta31 and tell us if it fixes the issue for you?

**Comment by alvesman** [comment]
Version 1.13.0-rc2-beta31 (9123)
Running OK on Hyper-V. When switching to Windows Containers:

Error response from daemon: open \\.\pipe\docker_engine_windows: The system cannot find the file specified.

   em Docker.Backend.DockerDaemonChecker.Check()
   em Docker.Backend.ContainerEngine.Windows.DoStart(Settings settings)
   em Docker.Backend.ContainerEngine.Windows.Start(Settings settings)
   em Docker.Core.Pipe.NamedPipeServer.<>c__DisplayClass8_0.<Register>b__0(Object[] parameters)
   em Docker.Core.Pipe.NamedPipeServer.RunAction(String action, Object[] parameters)

**Comment by rneugeba** [comment]
@alvesman thanks for trying anf for the logs. The important portion is:
```
[09:10:44.043][WindowsDockerDaemon][Info   ] Starting C:\Program Files\Docker\Docker\Resources\dockerd.exe -H npipe:////./pipe/docker_engine_windows
[09:10:44.048][WindowsDockerDaemon][Info   ] Started
[09:10:44.050][ApiProxy       ][Info   ] Killing existing com.docker.proxy with PID 13600
[09:10:44.055][ApiProxy       ][Info   ] Starting C:\Program Files\Docker\Docker\Resources\com.docker.proxy.exe -daemonNamedPipe=\\.\pipe\docker_engine_windows
[09:10:44.055][ApiProxy       ][Error  ] Process died
[09:10:44.059][ApiProxy       ][Info   ] Started
[09:10:44.100][ApiProxy       ][Info   ] docker proxy (on deprecated port): ready
[09:10:44.100][ApiProxy       ][Info   ] docker proxy: ready
[09:10:44.112][WindowsDockerDaemon][Info   ] time="2016-12-05T09:10:44.097129800Z" level=info msg="Windows default isolation mode: hyperv"
[09:10:44.118][WindowsDockerDaemon][Info   ] time="2016-12-05T09:10:44.118627300Z" level=info msg="Graph migration to content-addressability took 0.00 seconds"
[09:10:44.119][WindowsDockerDaemon][Info   ] time="2016-12-05T09:10:44.118627300Z" level=info msg="Loading containers: start."
[09:10:44.194][WindowsDockerDaemon][Info   ] Error starting daemon: Error initializing network controller: Error creating default network: HNS failed with error : O parâmetro está incorreto.
[09:10:44.196][DockerDaemonChecker][Error  ] Docker daemon is not running
[09:10:44.199][WindowsDockerDaemon][Error  ] Process died
[09:10:44.233][ApiProxy       ][Info   ] proxy >> GET /_ping
[09:10:44.233][ApiProxy       ][Info   ] Dial name pipe  \\.\pipe\docker_engine_windows
```

We have since released another Beta (Beta32.1) which has an updated docker daemon. Could you try that as well?

Also, If that also fails, Microsoft provides a handy script to debug Window container issues [here](https://github.com/Microsoft/Virtualization-Documentation/tree/live/windows-server-container-tools/Debug-ContainerHost), Could you try that as well as @JMesser81 s [outstanding PR](https://github.com/Microsoft/Virtualization-Documentation/pull/513) which adds additional networking related debug and also has a script to fix some networking related issues.

**Comment by alvesman** [comment]
Version 1.13.0-rc3-beta33 (9442)
Running OK on Hyper-V. When switching to Windows Containers:

Docker daemon failed with message:
time="2016-12-15T15:11:29.118169900Z" level=info msg="Windows default isolation mode: hyperv" 
time="2016-12-15T15:11:29.140168400Z" level=info msg="Graph migration to content-addressability took 0.00 seconds" 
time="2016-12-15T15:11:29.140168400Z" level=info msg="Loading containers: start." 
Error starting daemon: Error initializing network controller: Error creating default network: HNS failed with error : O parÃ¢metro estÃ¡ incorreto. **_(incorrect parameter)_**

   em Docker.Backend.ContainerEngine.Windows.DoStart(Settings settings)
   em Docker.Backend.ContainerEngine.Windows.Start(Settings settings)
   em Docker.Core.Pipe.NamedPipeServer.<>c__DisplayClass8_0.<Register>b__0(Object[] parameters)
   em Docker.Core.Pipe.NamedPipeServer.RunAction(String action, Object[] parameters)
[Debug-ContainerHost.ps1-output.zip](https://github.com/docker/for-win/files/654984/Debug-ContainerHost.ps1-output.zip)
[log.zip](https://github.com/docker/for-win/files/654985/log.zip)

**Comment by jornh** [comment]
@alvesman to get around this error in your Debug-ContainerHost.ps1.txt dump (which I also have experienced):
```
Describing Container network is created
 [-] Error occurred in Describe block 1.04s
   RuntimeException: Cannot index into a null array.
   at <ScriptBlock>, E:\TEMP\Debug-ContainerHost.ps1: line 128
```
You can try `New-ContainerNetwork -Name nat -Mode NAT –subnetprefix 10.0.76.0/24` in an elevated PowerShell. If that doesn't do it try the repair script as suggested in https://github.com/docker/for-win/issues/178#issuecomment-266272334 above.

**Comment by alvesman** [comment]
PS C:\Temp> New-ContainerNetwork -Name nat -Mode NAT –subnetprefix 10.0.75.0/24
New-ContainerNetwork : Object already exists
At line:1 char:1
+ New-ContainerNetwork -Name nat -Mode NAT –subnetprefix 10.0.75.0/24
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [New-ContainerNetwork], VirtualizationException
    + FullyQualifiedErrorId : OperationFailed,Microsoft.Containers.PowerShell.Cmdlets.NewContainerNetwork

Output of Debug-ContainerHost.ps1 in Debug-ContainerHost.txt
[Debug-ContainerHost.txt](https://github.com/docker/for-win/files/657104/Debug-ContainerHost.txt)

**Comment by alvesman** [comment]
Problem solved with Version 1.13.0-rc5-beta35 (9606)

**Comment by rneugeba** [comment]
@alvesman glad it's working for you with the update. I'm closing this. @HuanhuanSunMSFT you reported this issue quite a while ago. If you still have the issue on the recent Beta, please re-open.
thanks
